database/query/repositories.sql

```diff
@@ -28,7 +28,9 @@ SELECT * FROM repositories WHERE provider = $1 AND repo_id = $2 AND project_id =
 -- name: ListRepositoriesByProjectID :many
 SELECT * FROM repositories
 WHERE provider = $1 AND project_id = $2
-ORDER BY repo_name;
+  AND (repo_id > sqlc.narg('repo_id') OR sqlc.narg('repo_id') IS NULL)
+ORDER BY project_id, provider, repo_id
+LIMIT sqlc.narg('limit');
```

Comment:
Yes, I considered it. The main reason for not going with offset-limit-based pagination is that it is slower than cursor-based pagination, and records can be [potentially](https://developer.zendesk.com/documentation/api-basics/working-with-data/understanding-the-limitations-of-offset-pagination/) skipped if we are paginating using offset and limit.

The downside to cursor-based pagination is that we cannot go to any random page directly. Instead, it would be like an infinite scroll where more records can be fetched as the user scrolls through.

Reply:
I'm actually fine with this approach, only thing that threw me off was the `project + provider + repo_id` index, I would have thought `project + repo` was sufficient. But I'm also flexible and willing to iterate.

Reply:
@evankanderson @Vyom-Yadav thank you for the explanation about the cursor based search. I don't have any more comments, so feel free to ack once you're happy with the PR

Reply:
> only thing that threw me off was the project + provider + repo_id index, I would have thought project + repo was sufficient.

The reason for that is we have to make sure the cursor is valid. A `project + repo` cursor would work right now as github is the only provider, but if we had gitlab too (for eg), then differentiating the cursor would not be possible. Basically, we have to encode the previous query parameters in the cursor for correct retrieval.

Reply:
That's a good point!